src/net_processing.cpp

```diff
@@ -389,6 +397,10 @@ limitedmap<uint256, std::chrono::microseconds> g_already_asked_for GUARDED_BY(cs
 /** Map maintaining per-node state. */
 static std::map<NodeId, CNodeState> mapNodeState GUARDED_BY(cs_main);
 
+/** In-memory cache of all BIP157 compact filter checkpoints for the active chain. */
+static std::vector<std::pair<const CBlockIndex*, uint256>> active_chain_cf_headers;
```

Comment:
This is a nontrivial amount of memory. Can we not keep some limited cache for blocks near the tip and look up other ones?